configuration/src/test/java/io/camunda/configuration/SecondaryStorageTest.java

```diff
@@ -0,0 +1,179 @@
+      assertThat(exporterConfiguration.getConnect().getUrl()).isEqualTo(expectedUrl);
+    }
+
+    @Test
+    void testCamundaSearchEngineConnectProperties() {
+      assertThat(searchEngineConnectProperties.getType().toLowerCase()).isEqualTo("elasticsearch");
+      assertThat(searchEngineConnectProperties.getUrl()).isEqualTo("http://expected-url:4321");
+    }
+  }
+
+  @Nested
+  @TestPropertySource(
+      properties = {
+        // type
+        "camunda.data.secondary-storage.type=elasticsearch",
+        "camunda.database.type=elasticsearch",
+        "camunda.operate.database=elasticsearch",
+        "camunda.tasklist.database=elasticsearch",
+        // url
+        "camunda.data.secondary-storage.elasticsearch.url=http://matching-url:4321",
+        "camunda.database.url=http://matching-url:4321",
+        "camunda.tasklist.elasticsearch.url=http://matching-url:4321",
+        "camunda.operate.elasticsearch.url=http://matching-url:4321"
+      })
+  class WithNewAndLegacySet {
+    final OperateProperties operateProperties;
+    final TasklistProperties tasklistProperties;
+    final BrokerBasedProperties brokerBasedProperties;
+    final SearchEngineConnectProperties searchEngineConnectProperties;
+
+    WithNewAndLegacySet(
+        @Autowired final OperateProperties operateProperties,
+        @Autowired final TasklistProperties tasklistProperties,
+        @Autowired final BrokerBasedProperties brokerBasedProperties,
+        @Autowired final SearchEngineConnectProperties searchEngineConnectProperties) {
+      this.operateProperties = operateProperties;
+      this.tasklistProperties = tasklistProperties;
+      this.brokerBasedProperties = brokerBasedProperties;
+      this.searchEngineConnectProperties = searchEngineConnectProperties;
+    }
+
+    @Test
+    void testCamundaDataSecondaryStorageOperateProperties() {
+      final DatabaseType expectedOperateDatabaseType = DatabaseType.Elasticsearch;
+      final String expectedUrl = "http://matching-url:4321";
+
+      assertThat(operateProperties.getDatabase()).isEqualTo(expectedOperateDatabaseType);
+      assertThat(operateProperties.getElasticsearch().getUrl()).isEqualTo(expectedUrl);
+    }
+
+    @Test
+    void testCamundaDataSecondaryStorageTasklistProperties() {
+      final String expectedTasklistDatabaseType = "elasticsearch";
+      final String expectedUrl = "http://matching-url:4321";
+
+      assertThat(tasklistProperties.getDatabase()).isEqualTo(expectedTasklistDatabaseType);
+      assertThat(tasklistProperties.getElasticsearch().getUrl()).isEqualTo(expectedUrl);
+    }
+
+    @Test
+    void testCamundaDataSecondaryStorageCamundaExporterProperties() {
+      final String expectedType = "elasticsearch";
+      final String expectedUrl = "http://matching-url:4321";
+
+      final ExporterCfg camundaExporter =
+          UnifiedConfigurationHelper.getCamundaExporter(brokerBasedProperties);
```

Comment:
:x: Since this method is only needed for testing, move it out of the helper class to this test.